starboard/doc/networking_performance.md

```diff
@@ -0,0 +1,85 @@
+It is strongly encouraged for platforms to include the
+`CobaltExtensionSocketReceiveMultiMsgApi` extension.
+
+Starting in Cobalt 25.tls.20, a significant networking performance
+improvement can be achieved on platforms that implement the
+extension.
+
+The extension allows a platform to provide an API to read multiple
+UDP packets in a single call.  When on a Linux kernel, the `recvmmsg`
+system call can be used and results in a significantly reduced amount
+of CPU usage and corresponding increase in networking throughput.
+This not only makes the application more responsive, but also
+improves the video quality that the devices can play.
+
+A functioning implementation for Linux kernel based devices is
+included in the 25.lts.20 source and can be added to a platform with
+the changes below.  The changes are present in the source for the
+Android and Linux targets.
+
+To add the extension to a platform, (1) add the four source files
+with the extension's implementation to the platform and (2) expand
+`SbSystemGetExtension` to return the extension:
+
+1.  In the platform's `BUILD.gn`: 
+    Add the extension's implementation to the
+    `static_library("starboard_platform")` section.
+
+```
+diff --git a/starboard/linux/shared/BUILD.gn b/starboard/linux/shared/BUILD.gn
+index 6c01aecfe25..ddb3a4f666e 100644
+--- a/starboard/linux/shared/BUILD.gn
++++ b/starboard/linux/shared/BUILD.gn
+@@ -203,6 +203,10 @@ static_library("starboard_platform_sources") {
+     "//starboard/shared/posix/socket_join_multicast_group.cc",
+     "//starboard/shared/posix/socket_listen.cc",
+     "//starboard/shared/posix/socket_receive_from.cc",
++    "//starboard/shared/posix/socket_receive_multi_msg_internal.cc",
++    "//starboard/shared/posix/socket_receive_multi_msg_internal.h",
++    "//starboard/shared/posix/socket_receive_multi_msg.cc",
++    "//starboard/shared/posix/socket_receive_multi_msg.h",
+     "//starboard/shared/posix/socket_resolve.cc",
+     "//starboard/shared/posix/socket_send_to.cc",
+     "//starboard/shared/posix/socket_set_broadcast.cc",
+```
+
+2.  In the platform's `system_get_extensions.cc`: 
+    Add two includes and a three line if statement.
+
+```
+diff --git a/starboard/linux/shared/system_get_extensions.cc b/starboard/linux/shared/system_get_extensions.cc
```

Comment:
I wouldn't copy code in here - just a perma-link to the implementation is fine ( can use a tag link or such to ensure it stays around )

Reply:
Github gives no option to perma-link to the diff of only a single file, and doesn't allow a link with multiple lines that are not together. The PR also includes the implementation, but using it only takes edits to these two files.

The files themselves already have many other lines of code. A developer looking to implement would greatly benefit from a diff versus the entire implementaton PR or the entire files.

Reply:
Changed to a patch command, with shorter context.